go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -22,6 +22,92 @@ func deleteVIPsFromAllOVNBalancers(vips sets.String, name, namespace string) err
 	err := deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)
 	if err != nil {
 		return errors.Wrapf(err, "Failed to delete vips from ovn balancers %s %s", name, namespace)
+		
+type lbEndpoints struct {
+	IPs  []string
+	Port int32
+}
+
+func getLocalLbEndpoints(slices []*discovery.EndpointSlice, svcPort v1.ServicePort, family v1.IPFamily, gatewayRouter string) lbEndpoints {
```

Comment:
this function duplicates almost all of the code from GetLbEndpoints AFAICT. Instead of doing this, I would recommend just making getLBEndpoints return two things: LbEndpoints and LbEndpointsPerNode map[nodeName]LbEndpoints

Then we can call getLbEndpoints once in services_controller and get both pieces of information